sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -1282,7 +1282,7 @@ def get_module_fw_info(self):
             return {'status': False, 'info': "CDB Not supported", 'result': None}
 
         # get fw info (CMD 0100h)
-        rpllen, rpl_chkcode, rpl = self.cdb.get_fw_info()
+        rpllen, rpl_chkcode, rpl = self.cdb.get_fw_info()['rpl']
```

Comment:
@xinyulin Can we check for the value of `self.cdb.get_fw_info()['status']` and take appropriate action?

Reply:
@mihirpat1 I have added the check and corresponding actions in the get_module_fw_mgmt_feature() and get_module_fw_info() functions. Could you please review them again? Thanks.